ui/src/app/applications/components/application-deployment-history/revision-metadata-rows.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import {Timestamp} from '../../../shared/components/timestamp';
+import * as models from '../../../shared/models';
+import {services} from '../../../shared/services';
+
+interface Props {
+    applicationName: string;
+    revision: string;
+}
+
+interface State {
+    revisionMetadata?: models.RevisionMetadata;
+    error?: Error;
+}
+
+export class RevisionMetadataRows extends React.Component<Props, State> {
+    constructor(props: Props) {
+        super(props);
+        this.state = {};
+    }
+
+    public componentDidMount() {
+        services.applications.revisionMetadata(this.props.applicationName, this.props.revision)
+            .then((value) => this.setState({revisionMetadata: value}))
+            .catch((e) => this.setState({error: e}));
+    }
+
+    public render() {
+        const revisionMetadata = this.state.revisionMetadata;
```

Comment:
Sorry, did not notice it first time. Using DataLoader you can get rid of state and convert component to stateless function component:

```
<DataLoader input={props} load={(input) => services.applications.revisionMetadata(input.applicationName, input.revision)}>
{(revisionMetadata) => (
  <div>
  ...
  </div>
)}
</DataLoader>
```